### ISSUE TITLE
(Feature) Autocomplete and match existing objects

### ISSUE THREAD

**Issue by chriskl** [archimatetool/archi#340]
Archi does a great job of noticing when you've drawn in a relationship that already exists in the model and prompts you to re-use it.  However, it doesn't do this for objects.

What would be great is:

1. As you type the name of a newly created object, it offers autocomplete suggestions from existing objects in the model.
 - If you select an autocomplete entry it drops an instance of that existing object in place instead of the new one
 - if you continue typing and the name is still the name of an existing object, when you hit enter or blur the object, a popup dialog offers you the choice of (a) creating a new object with an identical name to the existing or (b) choose to use instead one from a list of existing objects with that name

2. An object 'merge' function. Select 2 or more objects of the same type in the Model browser and merge them into one, all views are updated

3. A more general 'Find objects with identical names'.  That will let you find mistakes in your model easily and offer a quick way to merge them.

In general, it's likely a mistake to have identically named objects of the same type in a model, but this mistake is really easy to make unless one has an encyclopaedic memory of the model.  And once you've made the mistake, it's really hard to fix it.

**Comment by tjsiefring** [comment]
The auto-complete would be great.  I can see a need for #2 & 3 but #1 I think would be far more useful to me.

**Comment by kyrreb** [comment]
This suggested feature would be a great improvement to prevent duplicating elements in the repository. Especially for teams of architects with more than one or two people working in the same repository, this would help in keeping the repo clean and free of duplicates and clutter.

**Comment by rolfmadsen** [comment]
I agree that 1 would be really usefull to avoid duplicates especially in large models.

I have experimented with a plugin https://github.com/rolfmadsen/archimatetool-duplicate-merge-plugin to solve 2, which works fairly well. The motivation was a recent consolidation of multiple models into one where duplicates created a mess.

It works by selecting objects of the same type and name in the model explorer, right click and select "Merge Duplicate Elements..." at the bottom of the menu.

Features
- Element Merging: Consolidates multiple elements of the same type and name into a single target element.
- Relationship Migration: Automatically moves all relationships from duplicates to the target element.
- Diagram Consolidation: Detects if duplicate elements appear on the same diagram, migrates connections to the target box, and removes the redundant box.
- Safe Connection Handling: Uses proper bi-directional updates to prevent model corruption and "hanging" connections.

NB. Please note that the latest commit is 7 months old so it has not been testet with the most recent versions of Archi.
